plugins/processors/lookup/README.md

```diff
@@ -0,0 +1,140 @@
+# Lookup Processor Plugin
+
+The Lookup Processor allows to use one or more files containing a lookup-table
+for annotating incoming metrics. The lookup is _static_ as the files are only
+used on startup. The main use-case for this is to annotate metrics with
+additional tags e.g. dependent on their source. Multiple tags can be added
+depending on the lookup-table _files_.
+
+The lookup key can be generated using a Golang template with the ability to
+access the metric name via `{{.Name}}`, the tag values via `{{.Tag "mytag"}}`,
+with `mytag` being the tag-name and field-values via `{{.Field "myfield"}}`,
+with `myfield` being the field-name. Non-existing tags and field will result
+in an empty string or `nil` respectively. In case the key cannot be found, the
+metric is passed-trough unchanged. By default all matching tags are added and
+existing tag-values are overwritten.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Lookup a key derived from metrics in a static file
+[[processors.lookup]]
+  ## List of files containing the lookup-table
+  files = ["path/to/lut.json", "path/to/another_lut.json"]
+
+  ## Format of the lookup file(s)
+  ## Available formats are:
+  ##    json               -- JSON file with 'key: {tag-key: tag-value, ...}' mapping
+  ##    csv_key_name_value -- CSV file with 'key,tag-key,tag-value,...,tag-key,tag-value' mapping
+  ##    csv_key_values     -- CSV file with a header and 'key,tag-value,...,tag-value' mapping
```

Comment:
Based on this alone, I'm not sure I totally understand what `csv_key_values` is and how it would look. Can you check my understanding:

`json` makes sense, you are looking for a specific key and you take that entire dictionary.

`csv_key_name_values` is the same, you look for the line with that key, and take the key-value pairs

`csv_key_values` based on the below, the header includes the tag-names and then the rows contain the tag values?

Maybe all that is missing is saying the header includes the tag-name?

Reply:
Your understanding is correct. Will add change to
```suggestion
  ##    csv_key_values     -- CSV file with a header containing '-,tag-name,...,tag-name' and rows with 'key,tag-value,...,tag-value' mapping
```